src/v/kafka/server/group_manager.cc

```diff
@@ -131,8 +131,8 @@ ss::future<> group_manager::stop() {
 
 void group_manager::detach_partition(const model::ntp& ntp) {
     klog.debug("detaching group metadata partition {}", ntp);
-    ssx::spawn_with_gate(_gate, [this, ntp]() -> ss::future<> {
-        auto it = _partitions.find(ntp);
+    ssx::spawn_with_gate(_gate, [this, _ntp{ntp}]() -> ss::future<> {
```

Comment:
when a lambda that is a coroutine suspends it will destroy the captures. this is still a capture just with a different name. the capture needs to be copied onto the stack / coroutine frame.

```
[this, _ntp{ntp}] {
  auto ntp{_ntp};
  auto it = _partitions.find(ntp);
```